src/main/java/frc/utils/pid/PIDConfig.java

```diff
@@ -0,0 +1,81 @@
+package frc.utils.pid;
+
+import edu.wpi.first.math.Pair;
+import java.util.HashMap;
+import java.util.function.BiConsumer;
+
+/**
+ * Stores config values for a {@link com.revrobotics.SparkPIDController}. Contains some helper
+ * methods for abstraction.
+ *
+ * @see frc.utils.pid.PIDParam
+ */
+public class PIDConfig {
+
+  // Stores mapping of PIDParam to the parameter key and value.
+  private HashMap<PIDParam, Pair<String, Double>> configValues;
+
+  /**
+   * Constructs a new PIDConfig.
+   *
+   * @param PGainKey
+   * @param PGainValue
+   * @param IGainKey
+   * @param IGainValue
+   * @param DGainKey
+   * @param DGainValue
+   * @param IZoneKey
+   * @param IZoneValue
+   * @param FFGainKey
+   * @param FFGainValue
+   * @param MaxOutputKey
+   * @param MaxOutputValue
+   * @param MinOutputKey
+   * @param MinOutputValue
+   */
+  public PIDConfig(
+      String PGainKey,
+      double PGainValue,
+      String IGainKey,
+      double IGainValue,
+      String DGainKey,
+      double DGainValue,
+      String IZoneKey,
+      double IZoneValue,
+      String FFGainKey,
+      double FFGainValue,
+      String MaxOutputKey,
+      double MaxOutputValue,
+      String MinOutputKey,
+      double MinOutputValue) {
+    this.configValues.put(PIDParam.P, Pair.of(PGainKey, PGainValue));
+    this.configValues.put(PIDParam.I, Pair.of(IGainKey, IGainValue));
+    this.configValues.put(PIDParam.D, Pair.of(DGainKey, DGainValue));
+    this.configValues.put(PIDParam.I_ZONE, Pair.of(IZoneKey, IZoneValue));
+    this.configValues.put(PIDParam.FF, Pair.of(FFGainKey, FFGainValue));
+    this.configValues.put(PIDParam.MAX_OUTPUT, Pair.of(MaxOutputKey, MaxOutputValue));
+    this.configValues.put(PIDParam.MIN_OUTPUT, Pair.of(MinOutputKey, MinOutputValue));
+  }
+
+  /** Gets the key and value for the corresponding {@link frc.utils.pid.PIDParam}. */
+  public Pair<String, Double> get(PIDParam param) {
```

Comment:
Can you specify this as nullable, since we don't always need an izone or feedforward specified for every pid controller. Only p,i and d gains are strictly needed.

Reply:
Will do, but is nullable better than setting a dummy value upon config initialization? I'm wondering if it's better to keep it null-safe.